twisted/internet/asyncioreactor.py

```diff
@@ -171,18 +171,18 @@ def addWriter(self, writer):
                                               self._readOrWrite, writer,
                                               False)
             self._writers[writer] = fd
-        except IOError as e:
+        except PermissionError:
             self._unregisterFDInAsyncio(fd)
-            if e.errno == errno.EPERM:
-                # epoll(7) doesn't support certain file descriptors,
-                # e.g. filesystem files, so for those we just poll
-                # continuously:
-                self._continuousPolling.addWriter(writer)
-            else:
-                raise
-        except OSError as e:
+            # epoll(7) doesn't support certain file descriptors,
+            # e.g. filesystem files, so for those we just poll
+            # continuously:
+            self._continuousPolling.addWriter(writer)
+        except BrokenPipeError:
             # The kqueuereactor will raise this if there is a broken pipe
```

Comment:
yeah, i think i'm just still upset that `OSError` *is* `IOError`, and a verbose comment would make me less upset :)  

once the builds are green, i say merge!